### PR TITLE
Fixed `PARAMETER_NAME_IN_OUTER_LAMBDA`

### LINKED ISSUE #1251: Fix findings of `PARAMETER_NAME_IN_OUTER_LAMBDA`
After #1228 is merged; the new rule discovers ~34 violations in our code base

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- fixed positive `PARAMETER_NAME_IN_OUTER_LAMBDA`
- suppress false positive `PARAMETER_NAME_IN_OUTER_LAMBDA`

It closes #1251

### CODE REVIEW

File: examples/gradle-kotlin-dsl/src/main/kotlin/Test.kt [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 
     val Incorrect_Val = 5
 
+    /**
+     * @throws Exception
+     */
+    fun incorrectFunction() {
+        throw Exception()
+    }
 }
```
Comment (DrAlexD) [inline examples/gradle-kotlin-dsl/src/main/kotlin/Test.kt:15]: Maybe it's worth changing all Test files in examples?

File: diktat-analysis.yml [hunk 2]
```diff
@@ -448,7 +448,7 @@
   enabled: true
 # Checks that outer lambda has explicit parameter name
 - name: PARAMETER_NAME_IN_OUTER_LAMBDA
-  enabled: false
+  enabled: true
 # Checks that property in constructor doesn't contain comment
 - name: KDOC_NO_CONSTRUCTOR_PROPERTY
   enabled: true
```
Comment (DrAlexD) [inline diktat-analysis.yml:454]: Change all `diktat-analysis.yml` files?

**Review by DrAlexD — APPROVED** [review]

File: examples/gradle-kotlin-dsl/src/main/kotlin/Test.kt (hunk 1 above)
Comment (nulls) [reply examples/gradle-kotlin-dsl/src/main/kotlin/Test.kt:15]: did it by mistake, will revert

File: diktat-analysis.yml (hunk 2 above)
Comment (nulls) [reply diktat-analysis.yml:454]: nope, will update the rest too

File: examples/gradle-kotlin-dsl/src/main/kotlin/Test.kt (hunk 1 above)
Comment (nulls) [reply examples/gradle-kotlin-dsl/src/main/kotlin/Test.kt]: done

File: diktat-analysis.yml (hunk 2 above)
Comment (nulls) [reply diktat-analysis.yml:454]: just to clarify:
- `diktat-analysis.yml` in root -- is our config file which is used for checking diktat
- `diktat-rules/src/main/resources/diktat-analysis.yml` and `diktat-rules/src/main/resources/diktat-analysis-huawei.yml` -- is config files which are provided in `diktat-rules` as resources as based configuration for checking our consumers

**Pull request merged**